main/config.c

```diff
@@ -2052,24 +2107,24 @@ static int process_text_line(struct ast_config *cfg, struct ast_category **cat,
 			c = NULL;
 		}
 		if (!strcasecmp(cur, "include")) {
-			do_include = 1;
+			include_type = 0;
 		} else if (!strcasecmp(cur, "tryinclude")) {
-			do_include = 1;
-			try_include = 1;
+			include_type = -1;
 		} else if (!strcasecmp(cur, "exec")) {
 			if (!ast_opt_exec_includes) {
 				ast_log(LOG_WARNING, "Cannot perform #exec unless execincludes option is enabled in asterisk.conf (options section)!\n");
 				return 0;	/* XXX is this correct ? or we should return -1 ? */
 			}
+			include_type = 1;
 		} else {
 			ast_log(LOG_WARNING, "Unknown directive '#%s' at line %d of %s\n", cur, lineno, configfile);
 			return 0;	/* XXX is this correct ? or we should return -1 ? */
 		}
 
 		if (c == NULL) {
 			ast_log(LOG_WARNING, "Directive '#%s' needs an argument (%s) at line %d of %s\n",
-					do_include ? "include / tryinclude" : "exec",
-					do_include ? "filename" : "/path/to/executable",
+					(include_type == 0) ? "include" : (include_type < 0) ? "tryinclude" : "exec",
```

Comment:
I would ditch the parentheses around include_type == 0
I get your point about clarity, but there's no ambiguity in cases like that and my experience with the Asterisk codebase is avoiding unnecessary parentheses when not needed. I've gotten called out for that before.
Keeping for the ternary args is fine.